### DISCUSSION TITLE
issue: setFocus does not work with <Controller />

### DISCUSSION THREAD

**Discussion by Crismon96** [react-hook-form/react-hook-form#10137] [Q&A] [answered]
### Version Number

7.41.0

### Codesandbox/Expo snack

https://codesandbox.io/s/react-hook-form-with-material-ui-forked-7nvhd8?file=/src/index.js

### Steps to reproduce

1. Open Sandbox
2. See that MUI Textfield with Controller is not focused.
3. In contrast the same works for the bar html "firstName" input.

In my production code I work with useFieldArray and the "shouldFocus" parameter of append doesn't work as well with has likely to do with this, more underlying issue. 

### Expected behaviour

The inputRef of the MUI textfield should be focused when using focus methods like "setFocus" or the parameter "shouldFocus: true".

### What browsers are you seeing the problem on?

Firefox, Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by leapful** [comment, marked as answer]
@Crismon96 

Please use read carefully the documentation of MUI TextField, prop `inputRef` should be used to register reference element instead of `ref` prop.

Working demo: https://codesandbox.io/s/react-hook-form-with-material-ui-forked-s7b8ss?file=/src/index.js

**Reply by Crismon96** [reply]
Thanks for your super quick help and reply. Everything works great now.
